busybox/busybox.py

```diff
@@ -26,21 +26,23 @@ def harness(dut: Dut, ctx: TestContext, result: TestResult):
         parsed = dut.match.groupdict()
 
         if idx == 2:
-            msg.append("\t\t" + parsed["line"])
-            continue
+            line_msg = parsed["line"] + " "
+            # append extra message to the last test (if available)
+            if subresult and subresult.status == Status.FAIL:
+                subresult.msg += line_msg
+            else:
+                # message may also appear during the test
+                msg.append(line_msg)
 
-        if subresult:
-            # We ended processing test result and message
-            if msg and subresult.status == Status.FAIL:
+        if idx == 0:
+            status = Status.from_str(parsed["status"])
+
+            if status == Status.FAIL:
                 test_status = Status.FAIL
-                subresult.msg = "\n".join(msg)
-                msg = []
 
-            result.add_subresult_obj(subresult)
-            subresult = None
+            subresult = result.add_subresult(subname=parsed["name"], status=status, msg="".join(msg))
```

Comment:
Don't be afraid of newlines (I think we can use them instead of `" "` above also)

```suggestion
            subresult = result.add_subresult(subname=parsed["name"], status=status, msg="\n".join(msg))
```